silq/parameters/acquisition_parameters.py

```diff
@@ -1965,6 +1965,69 @@ def analyse(self, traces: Dict[str, Dict[str, np.ndarray]] = None):
         return results
 
 
+class EDSRParameter(NMRParameter):
+    """
+    Parameter for EDSR measurements based on NMR parameter and pulse sequence
+
+    Refer to NMRParameter for details. In addition to all properties copied from NMRParameter,
+    EDSRParameter includes analysis of electron readout right after EDSR(NMR) pulse during
+    NMR['post_pulse'] = DCPulse('read') that needs to be present in NMRPulseSequence.
+
+    Args:
+        Refer to NMRParameter
+    Parameters:
+        Refer to NMRParameter
+    """
+
+    def __init__(self, name: str = 'EDSR',
+                 names: List[str] = ['flips', 'flip_probability', 'up_proportions',
+                                     'state_probability', 'threshold_up_proportion',
+                                     'EDSR_up_proportion'],
+                 **kwargs):
+        super().__init__(name=name,
+                         names=names,
+                         **kwargs)
+
+    @property
+    def names(self):
+        names = super().names
+        names.append('EDSR_up_proportion')
+        return names
+
+    @names.setter
+    def names(self, names):
+        self._names = names
+
+    def analyse(self, traces: Dict[str, Dict[str, np.ndarray]] = None):
+        """
+        Reading out electron spin-up proportion after EDSR (NMR) pulse during 'read' DCPulse.
+
+        Returns:
+            (Dict[str, Any]): Dict containing:
+            * all results from NMRParameter
+            * **EDSR_up_proportion**: electron spin-up proportion right after EDSR (NMR) pulse.
+        """
+        results = super().analyse(traces)
+
+        # Extract points for read DC pulse after EDSR pulse
+        EDSR_read_traces_name = f"{self.NMR['post_pulse'].name}"
+        EDSR_read_traces = traces[EDSR_read_traces_name]['output']
+        EDSR_trace_points = EDSR_read_traces.shape[1]
+
+        self.EDSR_traces = np.zeros((self.samples, EDSR_trace_points))
+        self.EDSR_traces = EDSR_read_traces
```

Comment:
This line is also unnecessary. I don't think it's needed to have an additional `EDSR_traces` attribute since it can already be accessed via self.traces. Otherwise it would also make sense to add `ESR_traces` and `NMR_traces`

Reply:
corrected.